game-core/src/main/java/games/strategy/engine/framework/map/download/ContentReader.java

```diff
@@ -63,9 +63,7 @@ void downloadToFile(final String uri, final File file) throws IOException {
     checkNotNull(uri);
     checkNotNull(file);
 
-    try (FileOutputStream os = new FileOutputStream(file)) {
-      downloadAndApplyAction(
-          uri, is -> os.getChannel().transferFrom(Channels.newChannel(is), 0L, Long.MAX_VALUE));
-    }
+    downloadAndApplyAction(
+        uri, is -> Files.copy(is, file.toPath(), StandardCopyOption.REPLACE_EXISTING));
```

Comment:
@RoiEXLab are sure downloading multiple maps in parallel will have no issues?

Reply:
Well if it worked before without any issues, it will continue to work without any issues.
If you're concerned about the lambda, the only reason a lambda is being used here is to make it hard to misuse the passed input stream: The stream is closed by `downloadAndApplyAction`.
So the only real change here is that we
1. No longer use java.nio.channels
2. let java.nio.Files handle the outputstream logic for us, instead of creating a stream on our own.

Reply:
But in case you're not convinced I just downloaded 4 maps concurrently without any issues (when downloading, got some non 200 status codes from github before actually starting the download)

Reply:
Thanks for checking, wanted to be sure we were using unique temp files per download.

Reply:
I'm pretty sure that's still the case, but as mentioned already it shouldn't really matter regarding this change.

The only difference I found by the way when having a look at the jdk source code, is that `FileOutputStream` seems to overwrite or append existing files if they are present, whereas the nio variant straight up deletes any existing file before creating a new one when the REPLACE_EXISTING flag is present.